projects/bootloader/scripts/flash_application_code.py

```diff
@@ -0,0 +1,85 @@
+# pylint: skip-file
+"""This script handles the client side of the flash application code protocol."""
+
+# Import protobuf
+from protogen import flash_application_code_pb2 as pb2
+from can_datagram import Datagram, DatagramListener, DatagramSender
+import can
+import time
+import os
+import sys
+currentdir = os.path.dirname(os.path.realpath(__file__))
+parentdir = os.path.dirname(currentdir)
+sys.path.append(parentdir)
+
+
+class FlashApplication():
```

Comment:
It's great you're doing this in an object-oriented style! For consistency with other operations, please add a top-level `flash_application_code` function that wraps the whole protocol (which would mean instantiating FlashApplication and calling flash_protobuf and flash_application).